memcache/selector.go

```diff
@@ -33,6 +33,8 @@ type ServerSelector interface {
 	// should be shared onto.
 	PickServer(key string) (net.Addr, error)
 	Each(func(net.Addr) error) error
+	// Reresolve try to reresolve address for given ip
+	Reresolve(addr *staticAddr) error
```

Comment:
extending this interfaces breaks compatibility for everybody who's already implemented this interface. 

I don't think this is workable as-is.